src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -81,9 +81,9 @@ export class TableOfContents implements OnInit {
   }
 
   ngOnDestroy(): void {
-    this._routeSubscription.unsubscribe();
-    this._scrollSubscription.unsubscribe();
-    this._fragmentSubscription.unsubscribe();
+    this._routeSubscription && this._routeSubscription.unsubscribe();
+    this._scrollSubscription && this._scrollSubscription.unsubscribe();
+    this._fragmentSubscription && this._fragmentSubscription.unsubscribe();
```

Comment:
This might be a nice time to use `takeUntil` so you don't have to manage all the subscriptions

```ts
private _onDestroy = new Subject<void>();

constructor() {
  this._router.events
    .takeUntil(this._onDestroy)
    .subscribe();
}

ngOnDestroy() {
  this._onDestroy.next();
  this._onDestroy.complete();
}


```

Reply:
Done